source/_drafts/Bugpost-Disconnects-and-Reconnects.md

```diff
@@ -0,0 +1,98 @@
+
+```
+/**
+ * Retrigger the `waiting` event that was triggered by the {@link Tech}.
+ *
+ * @fires Player#waiting
+ * @listens Tech#waiting
+ * @private
+ */
+handleTechWaiting_() {
+  this.addClass('vjs-waiting');
+  /**
+   * A readyState change on the DOM element has caused playback to stop.
+   *
+   * @event Player#waiting
+   * @type {EventTarget~Event}
+   */
+  this.trigger('waiting');
+  this.one('timeupdate', () => this.removeClass('vjs-waiting'));
+}
+```
+
+After logging the events emitted by the player, the issue became clear. Some browsers would emit a `timeupdate` event immediatley after the `waiting` event.
+
+So `timeupdate` events themselves were not reliable, but what else could be used to make a better determination of if we're waiting? We checked `player.currentTime()` on the `timeupdate` events, and found that when we caught the last `timeupdate` event, the one after the `waiting` event, the player was at the same time as the `timeupdate` event prior to the `waiting` event (and the same time as when the `waiting` event was triggered).
+
+```
+`timeupdate` triggered, player.currentTime = 11
+`timeupdate` triggered, player.currentTime = 11.250
+`waiting` triggered, player.currentTime = 11.250
+`timeupdate` triggered, player.currentTime = 11.250
+```
+
+The fix was pretty easy. Simply record the player's time when it gets the `waiting` event, and instead of removing the `vjs-waiting` class on the next `timeupdate`, remove the `vjs-waiting` class on the next `timeupdate` that has a different time from the one we recorded.
+
+You can see the PR for that [here](https://github.com/videojs/video.js/pull/5533), and it is available in Video.js 7.4.0.
+
+## Resuming Playback after Disconnect
+
+With the loading spinner back, the next issue was to determine why, when the network reconnected, the video did not resume. First, we tested a few different pieces of content. The HLS source we tested did reconnect, but the DASH source didn't. Worse, when using the DASH source, Chrome's debug console froze, as it racked up request and console errors as fast as the computer could process them.
+
+### Console Crash
+
+Imagine your console filled with the following two messages repeated thousands of times:
+```
+GET https://dash.akamaized.net/akamai/bbb_30fps/bbb_a64k/bbb_a64k_8.m4a net::ERR_INTERNET_DISCONNECTED
+VIDEOJS: WARN: Problem encountered with the current HLS playlist. Trying again since it is the final playlist.
+```
+
+When your console freezes, debugging can be tough. It's like trying to watch a video when the microwave is on. So we tackled that issue first.
+
+Looking at the error message, we were retrying the last rendition over and over. That is expected behavior, as we added that in https://github.com/videojs/videojs-contrib-hls/pull/1039 (a long time ago). However, DASH uses a different playlist loader than HLS, and we never added the same logic for reloading the final DASH playlists.
```

Comment:
I would just say `as we added that a [long time ago](https://github.com/videojs/videojs-contrib-hls/pull/1039)`.